engines/sci/engine/script.cpp

```diff
@@ -125,74 +126,83 @@ void Script::load(int script_nr, ResourceManager *resMan, ScriptPatcher *scriptP
 		// RAMA: 70
 		//
 		// TODO: Remove this once such a mechanism is in place
-		if (script->size > 65535)
-			warning("TODO: SCI script %d is over 64KB - it's %d bytes long. This can't "
-			      "be fully handled at the moment", script_nr, script->size);
+		if (script->size() > 65535)
+			warning("TODO: SCI script %d is over 64KB - it's %lu bytes long. This can't "
+			      "be fully handled at the moment", script_nr, script->size());
 	}
 
 	uint extraLocalsWorkaround = 0;
-	if (g_sci->getGameId() == GID_FANMADE && _nr == 1 && script->size == 11140) {
+	if (g_sci->getGameId() == GID_FANMADE && _nr == 1 && script->size() == 11140) {
 		// WORKAROUND: Script 1 in Ocean Battle doesn't have enough locals to
 		// fit the string showing how many shots are left (a nasty script bug,
 		// corrupting heap memory). We add 10 more locals so that it has enough
 		// space to use as the target for its kFormat operation. Fixes bug
 		// #3059871.
 		extraLocalsWorkaround = 10;
 	}
-	_bufSize += extraLocalsWorkaround * 2;
+	bufSize += extraLocalsWorkaround * 2;
 
-	_buf = (byte *)malloc(_bufSize);
-	assert(_buf);
-
-	assert(_bufSize >= script->size);
-	memcpy(_buf, script->data, script->size);
+	SciSpan<byte> outBuffer = _buf->allocate(bufSize, script->name() + " buffer");
+	script->copyDataTo(outBuffer);
+	_script = _buf->subspan(0, scriptSize, script->name());
```

Comment:
Looks like more work needs to be done for this beyond just using the unpadded size—changing only `scriptSize` to `script->size()` causes at least EQ2 to get a bogus locals size on some scripts.